### PR TITLE
Introduce a completely new Header component

### PR DESCRIPTION
**Pull request by duranmla**
**What:**
Create a new version of the header that adjust to the latest designs

**Why:**
Closes https://app.asana.com/0/1168997577035609/1200120162044961
Design: https://www.figma.com/file/b2MmXqHw1I8DS5FOaQXRUY/Debt-Collective-Redesign?node-id=4970%3A1066

**How:**
- Refactor to match new designs
- Re-organize code and use SCSS for styling
- Remove unnecessary components and rely on Header, Menu, and Profile to achieve design
- Add a custom link component that exposes an event in order to allow the host app to use inner routing

Things I saw but leave behind:
- Box shadow on profile picture when inactive wasn't achieve due to conflicts on active states image undesired resize effect
- Box shadow on profile menu connector wasn't achieved as difficulties to follow container shadow
- Box shadow on the menu at the bottom looks odd due to multiple box-shadow used on the menu footer
- Box shadow on the menu header was ignored intentionally as doesn't seem coherent
- Custom link component doesn't support render `img` _(which currently affects to logo only)_

Things that need to be implemented in the next PRs:
- Header scroll animation to be scaled
- Hover style for links on desktop devices

**Notes:**
While many of these notes were commented on a design I still want to arise here:

- Search is not implemented as there is no clear behavior yet
- Hierarchy of box shadows doesn't seem to be coherent _(menu and header should have the same shadow as same level)_

**Pending for merge**
- [x] Map all links to actual routes
- [x] Make sure custom link component works properly if no handler is passed

**Media:**
_Design and component:_
https://www.loom.com/share/960c5adea76148d2ad6f417e711ea5f0

_Code walkthrough:_
https://www.loom.com/share/36528516d7d248c3a63107a0b427fc2a

### CODE REVIEW

**Comment by duranmla** [comment]
I have marked this PR as not ready for merge yet, because, I need to map the links on the profile menu to the actual values. I currently not sure about what is the correct link for "drafts" CC: @thiagodemellobueno

File: packages/header-component/src/components/dc-header/header.tsx [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+   */
+  @State() user?: User;
+
+  /**
+   * Wether or not the menu is displayed
+   */
+  @State() isMenuOpen: boolean;
+
+  /**
+   * Wether or not the profile menu is displayed
```
Comment (orlando) [inline packages/header-component/src/components/dc-header/header.tsx:48]: typo whether

File: packages/header-component/src/index.html [hunk 2]
```diff
@@ -15,30 +15,41 @@
...
 
     <!-- You can choose to render custom navigation, check the README.md file and restart the stencil server after change index.html -->
     <script>
       const header = document.querySelector("#dc-header");
-      header.addEventListener("userSynced", event => {
+      header.addEventListener("userSynced", (event) => {
         console.log("userSynced", event.detail);
       });
+
+      header.addEventListener("linkClicked", ({ detail }) => {
```
Comment (orlando) [inline packages/header-component/src/index.html:49]: are we able to listen to these events from outside the component?

**Comment by thiagodemellobueno** [comment]
> I have marked this PR as not ready for merge yet, because, I need to map the links on the profile menu to the actual values. I currently not sure about what is the correct link for "drafts" CC: @thiagodemellobueno
> 
> **Besides, I need to understand which links from the menu will remain**

I say kill drafts for now it's only in there because it was in there beforem

As for which will remain, which ones are in question? You mean main menu or user menu? I can fast track getting answers Monday/Tues

**Comment by duranmla** [comment]
The source of Truth for this is [in this link](https://docs.google.com/document/d/1RZrlizaSPB8OgsXnu6zKoSTr5mWnUlAPIVbmdY2Hz8g/edit)

File: packages/header-component/src/index.html (hunk 2 above)
Comment (duranmla) [reply packages/header-component/src/index.html:49]: Yes, the idea of `index.html` is to simulate a host application.

**Comment by duranmla** [comment]
As I don't see any obstacles and after the talk on Monday I will merge this PR. 🎉

**Pull request merged**